### PR TITLE
add note in LightningCLI docs that `--optimizer` must be given for `--lr_scheduler` to work

### LINKED ISSUE #17547: clarify behavior of --lr_scheduler in LightningCLI
### 📚 Documentation

hello, wanted to say that i love using lightning. i am currently using the `LightningCLI` object to train my models. i implemented `configure_optimizers` on my LightningModule subclass, and using the cli, i was hoping to change the learning rate scheduler but use the optimizer defined by my `configure_optimizers`. in other words, i was only passing `--lr_scheduler` and not `--optimizer`. after some playing around with parameters and browsing the source, i see that `--lr_scheduler` only has an effect if `--optimizer` is also passed.

would you be interested in a pr that clarifies the `--help` text that `--lr_scheduler` requires `--optimizer`?

cc @borda @carmocca @mauvilsa

### PR DESCRIPTION
**Pull request by kaczmarj**
## What does this PR do?

This PR clarifies the documentation for `LightningCLI` regarding `--lr_scheduler`. Specifically, it adds the following:

> Please note that `--optimizer` must be added for `--lr_scheduler` to have an effect.

 It also adds `--optimizer=Adam` to examples of `--lr_scheduler` to reinforce the notion that both options must be provided.

Fixes #17547 

<details>
  <summary><b>Before submitting</b></summary>

- Was this **discussed/agreed** via a GitHub issue? (not for typos and docs) -- Yes
- [x] Did you read the [contributor guideline](https://github.com/Lightning-AI/lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- Did you make sure to **update the documentation** with your changes? (if necessary) -- Yes
- Did you write any **new necessary tests**? (not for typos and docs) -- NA
- [x] Did you verify new and **existing tests pass** locally with your changes?
- Did you list all the **breaking changes** introduced by this pull request? -- NA
- Did you **update the CHANGELOG**? (not for typos, docs, test updates, or minor internal changes/refactors) -- NA

</details>

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing, make sure you have read the [review guidelines](https://github.com/Lightning-AI/lightning/wiki/Review-guidelines). In short, see the following bullet-list:

<details>
  <summary>Reviewer checklist</summary>

- [ ] Is this pull request ready for review? (if not, please submit in draft mode)
- [ ] Check that all items from **Before submitting** are resolved
- [ ] Make sure the title is self-explanatory and the description concisely explains the PR
- [ ] Add labels and milestones (and optionally projects) to the PR so it can be classified

</details>

### CODE REVIEW

**Review by carmocca — APPROVED** [review]
Thanks!

**Review by mauvilsa — APPROVED** [review]

**Review by Borda — APPROVED** [review]
